src/main/java/in/org/projecteka/hiu/consent/ConsentService.java

```diff
@@ -87,15 +104,44 @@ public ConsentService(HiuProperties hiuProperties,
     }
 
     public Mono<Void> createRequest(String requesterId, ConsentRequestData consentRequestData) {
+        var gatewayRequestId = UUID.randomUUID();
         return validateConsentRequest(consentRequestData)
-                .then(sendConsentRequestToGateway(requesterId, consentRequestData));
+                .then(sendConsentRequestToGateway(requesterId, consentRequestData, gatewayRequestId));
+    }
+
+    public Mono<Map<String, String>> handlePatientConsentRequest(String requesterId, PatientConsentRequest consentRequest) {
+        Map<String, String> response = new HashMap<>();
+        return Flux.fromIterable(consentRequest.getHipIds())
+                .flatMap(hipId -> buildConsentRequest(requesterId).flatMap(consentRequestData -> {
+                    var patientRequestId = UUID.randomUUID();
```

Comment:
Rename this to dataRequestId as we discussed.

Reply:
done